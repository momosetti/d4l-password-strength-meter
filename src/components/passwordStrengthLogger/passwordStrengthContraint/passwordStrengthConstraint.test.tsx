import React from "react";
import { screen, render } from "@testing-library/react";
import "jest-styled-components";
import PasswordStrengthConstraint from "./index";

describe("test <PasswordStrenghtConstraint/>", () => {
  test("render <PasswordStrenghtConstraint/> correctly", () => {
    const { container } = render(
      <PasswordStrengthConstraint status={true} innerContent="hello world" />
    );

    expect(screen.getByText("hello world")).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("get an error indication when the status is false", () => {
    const { container } = render(
      <PasswordStrengthConstraint
        status={false}
        innerContent="capital letter"
      />
    );
    const iconEl = screen.getByTitle("icon");

    expect(iconEl).toHaveClass("ri-error-warning-fill");
    expect(container.firstChild).toHaveStyleRule("color", "var(--danger-red)");
  });

  test("get a valid indication when the status is true", () => {
    const { container } = render(
      <PasswordStrengthConstraint status={true} innerContent="capital letter" />
    );
    const iconEl = screen.getByTitle("icon");

    expect(iconEl).toHaveClass("ri-check-double-fill");
    expect(container.firstChild).toHaveStyleRule("color", "var(--green-ok)");
  });
});
