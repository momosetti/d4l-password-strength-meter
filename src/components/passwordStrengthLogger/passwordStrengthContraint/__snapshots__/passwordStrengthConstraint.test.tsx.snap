// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <PasswordStrenghtConstraint/> render <PasswordStrenghtConstraint/> correctly 1`] = `
.c0 {
  color: var(--green-ok);
}

.c0 i {
  vertical-align: middle;
  margin-right: 0.3em;
}

.c0 p {
  display: inline;
}

<li
  aria-hidden="true"
  class="c0"
>
  <i
    class="ri-check-double-fill"
    title="icon"
  />
  <p>
    hello world
  </p>
</li>
`;
